src/backend/distributed/commands/view.c

```diff
@@ -0,0 +1,349 @@
+ *
+ * The original list will not be touched, a new list will be created with only the objects
+ * in there.
+ */
+static List *
+FilterNameListForDistributedViews(List *viewNamesList, bool missing_ok)
+{
+	List *distributedViewNames = NIL;
+
+	List *qualifiedViewName = NULL;
+	foreach_ptr(qualifiedViewName, viewNamesList)
+	{
+		/*
+		 * Name of the view must be qualified before calling this function
+		 */
+		Assert(list_length(qualifiedViewName) == 2);
+
+		char *schemaName = strVal(linitial(qualifiedViewName));
+		char *viewName = strVal(lsecond(qualifiedViewName));
+
+		Oid schemaId = get_namespace_oid(schemaName, missing_ok);
+		Oid viewOid = get_relname_relid(viewName, schemaId);
+
+		if (!OidIsValid(viewOid))
+		{
+			continue;
+		}
+
+		ObjectAddress viewAddress = { 0 };
+		ObjectAddressSet(viewAddress, RelationRelationId, viewOid);
+
+		if (IsObjectDistributed(&viewAddress))
+		{
+			distributedViewNames = lappend(distributedViewNames, qualifiedViewName);
+		}
+	}
+	return distributedViewNames;
+}
+
+
+/*
+ * CreateViewDDLCommandsIdempotent returns a list of DDL statements (const char *) to be
+ * executed on a node to recreate the view addressed by the viewAddress.
+ */
+List *
+CreateViewDDLCommandsIdempotent(const ObjectAddress *viewAddress)
+{
+	StringInfo createViewCommand = makeStringInfo();
+
+	Oid viewOid = viewAddress->objectId;
+
+	char *viewName = get_rel_name(viewOid);
+	Oid schemaOid = get_rel_namespace(viewOid);
+	char *schemaName = get_namespace_name(schemaOid);
+
+	appendStringInfoString(createViewCommand, "CREATE OR REPLACE VIEW ");
+
+	AddQualifiedViewNameToCreateViewCommand(createViewCommand, viewOid);
+	AppendAliasesToCreateViewCommandForExistingView(createViewCommand, viewOid);
+	AddOptionsToCreateViewCommandForExistingView(createViewCommand, viewOid);
+	AddViewDefinitionToCreateViewCommand(createViewCommand, viewOid);
+
+	/* Add alter owner commmand */
+	StringInfo alterOwnerCommand = makeStringInfo();
+
+	char *viewOwnerName = TableOwner(viewOid);
+	char *qualifiedViewName = NameListToQuotedString(list_make2(makeString(schemaName),
+																makeString(viewName)));
+	appendStringInfo(alterOwnerCommand,
+					 "ALTER VIEW %s OWNER TO %s", qualifiedViewName,
+					 quote_identifier(viewOwnerName));
+
+	return list_make2(createViewCommand->data,
+					  alterOwnerCommand->data);
+}
+
+
+/*
+ * AppendAliasesToCreateViewCommandForExistingView appends aliases to the create view
+ * command for the existing view.
+ */
+static void
+AppendAliasesToCreateViewCommandForExistingView(StringInfo createViewCommand, Oid viewOid)
+{
+	/* Get column name aliases from pg_attribute */
+	ScanKeyData key[1];
+	ScanKeyInit(&key[0],
+				Anum_pg_attribute_attrelid,
+				BTEqualStrategyNumber, F_OIDEQ,
+				ObjectIdGetDatum(viewOid));
+
+	/* TODO: Check the lock */
+	Relation maprel = table_open(AttributeRelationId, AccessShareLock);
+	Relation mapidx = index_open(AttributeRelidNumIndexId, AccessShareLock);
```

Comment:
can we ever have dropped columns in views? We should be careful about that if dropped columns are possible (not sure)

Reply:
No it's not possible, dropping a column drops the view as well.

Reply:
just marking as related https://github.com/citusdata/citus/issues/5932, nothing actionable